src/shared/formatting/run-display-formatters.ts

```diff
@@ -64,3 +64,19 @@ export function formatDurationHoursMinutes(durationSeconds: number): string {
 export function formatTimestampDisplay(timestamp: Date): string {
   return formatDisplayShortDateTime(timestamp)
 }
+
+// ============================================================================
+// Game Speed Formatting
+// ============================================================================
+
+/**
+ * Format game speed with 'x' suffix.
+ * Removes unnecessary trailing zeros for cleaner display.
+ * Examples: 2.0 -> "2x", 2.5 -> "2.5x", 2.123 -> "2.123x"
+ * Game speed is the ratio of gameTime to realTime.
+ */
+export function formatGameSpeed(value: number): string {
```

Comment:
As per the discord convo, lets make this a lil more generic to handle multipliers.